Source/WebKit/UIProcess/Extensions/Cocoa/_WKWebExtensionDeclarativeNetRequestRule.mm

```diff
@@ -364,6 +368,23 @@ - (instancetype)initWithDictionary:(NSDictionary *)ruleDictionary errorString:(N
         }
     }
 
+    if (NSArray<NSString *> *requestMethods = _condition[declarativeNetRequestRuleConditionRequestMethodsKey]) {
+        if (!isArrayOfRequestMethodsValid(requestMethods)) {
+            if (outErrorString)
+                *outErrorString = [[NSString alloc] initWithFormat:@"Rule with id %ld is invalid. `requestMethods` must be non-empty and can only contain `get`, `head`, `options`, `trace`, `put`, `delete`, `post`, `patch`, and `connect`.", (long)_ruleID];
```

Comment:
Instead of duplicating the allowed methods, in theory we could create a helper method that returns the set of the allowed methods, and use it in the logging. I'm not sure if that needs to be done, just duplicating the list of them is a bit concerning in case we add any in the future.